tests/query/snql/test_query.py

```diff
@@ -31,16 +36,84 @@
                 FunctionCall(None, "plus", (Literal(None, 2), Literal(None, 7),),),
             ],
-            order_by=[
-                OrderBy(OrderByDirection.DESC, Column(None, None, "f")),
-                OrderBy(OrderByDirection.ASC, Column(None, None, "m")),
+            having=combine_or_conditions(
+                [
+                    binary_condition(
+                        None,
+                        "greater",
+                        Column(None, None, "times_seen"),
+                        Literal(None, 1),
+                    ),
+                    binary_condition(
+                        None,
+                        "equals",
+                        Column(None, None, "last_seen"),
+                        Literal(None, 2),
+                    ),
+                ]
+            ),
+            order_by=[OrderBy(OrderByDirection.DESC, Column(None, None, "f"))],
+        ),
+        id="Example 1",
+    ),
+    pytest.param(
+        "MATCH (blah) WHERE time_seen<3 COLLECT 4-5, 3*g(c), c HAVING b!=1 OR times_seen>1 AND last_seen=2",
+        Query(
+            {},
+            None,
+            selected_columns=[
+                SelectedExpression(
+                    "4-5",
+                    FunctionCall(None, "minus", (Literal(None, 4), Literal(None, 5),),),
+                ),
+                SelectedExpression(
+                    "3*g(c)",
+                    FunctionCall(
+                        None,
+                        "multiply",
+                        (
+                            Literal(None, 3),
+                            FunctionCall(None, "g", (Column(None, None, "c"),),),
+                        ),
+                    ),
+                ),
+                SelectedExpression("c", Column(None, None, "c"),),
             ],
+            condition=binary_condition(
+                None, "less", Column(None, None, "time_seen"), Literal(None, 3)
+            ),
+            having=combine_or_conditions(
+                [
+                    binary_condition(
+                        None, "notEquals", Column(None, None, "b"), Literal(None, 1)
+                    ),
+                    combine_and_conditions(
+                        [
+                            binary_condition(
+                                None,
+                                "greater",
+                                Column(None, None, "times_seen"),
+                                Literal(None, 1),
+                            ),
+                            binary_condition(
+                                None,
+                                "equals",
+                                Column(None, None, "last_seen"),
+                                Literal(None, 2),
+                            ),
+                        ]
+                    ),
+                ],
+            ),
         ),
-        id="Simple COLLECT, GROUPBY, ORDERBY clause example",
+        id="Example 2",
```

Comment:
I would suggest to add a few simpler tests that test some specific conditions:
1) multiple conditions in AND (more than 2). The other clauses can be trivial
2) multiple conditions in OR (more than 2). The other clauses can be trivial
3) Conditions with different types of literals (int, columns, strings)
4) Combination of AND and OR at multiple levels.

This approach will allow you to cover more cases without having a massive query each time to express with the AST.